### ISSUE TITLE
One backup per file

### ISSUE THREAD

**Issue by sabman3** [ansible/ansible#77042]
### Summary

Not exactly sure it can be done or really how. But can we get a feature where a backup is done only once per play? If checking for the presence of multiple lines in a file using lineinfile or something similar, only need to backup on the first change. Right now setting "backup: yes" creates a backup for each time the file is changed, when you only need the file backed up once. I understand that a backup can be done outside the module, but that creates a file even if nothing is changed. Maybe something in the main ansible app the keeps track of files already backed up during the play?

### Issue Type

Feature Idea

### Component Name

lineinfile, blockinfile, etc

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/blockinfile.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/blockinfile.py)
* [`lib/ansible/modules/lineinfile.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/lineinfile.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by bcoca** [comment]
Currently the backup is done inside the module/task and it knows nothing about previous or subsequent backups. 

One options is to create a backup task before the lineinfile with `copy: remote_src=yes` instead of using the builtin `backup`. 

Another option is you register each task and set `backup={{taskresult is not changed}}` to ensure subsequent tasks don't do their own backup if previous ones have already done so, since there are no backups done unless the change is needed.

**Comment by nitzmahone** [comment]
The backup mechanism is there for safety, so trying to have it be significantly smarter or overwriting existing files would probably hurt the safety it provides- we're unlikely to implement that. If you need different behavior, doing something explicit as @bcoca suggests is probably the best choice.

Closing as per above...
